aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,623 @@
+            let token_id = token::create_token_id_raw(gtoken.creator, gtoken.collection, token_name, property_version);
+            // check if this token already voted
+            assert!(!bucket_table::contains(&stats.no_votes, &token_id), error::invalid_argument(ETOKEN_ALREADY_VOTED));
+            assert!(!bucket_table::contains(&stats.yes_votes, &token_id), error::invalid_argument(ETOKEN_ALREADY_VOTED));
+
+            // this account owns the token
+            assert!(token::balance_of(signer::address_of(account), token_id) == 1, error::permission_denied(ENOT_OWN_THE_VOTING_DAO_TOKEN));
+            // this token is a member token
+            if (vote) {
+                stats.total_yes = stats.total_yes + 1;
+                bucket_table::add(&mut stats.yes_votes, token_id, voter_addr);
+            } else {
+                stats.total_no = stats.total_no + 1;
+                bucket_table::add(&mut stats.no_votes, token_id, voter_addr);
+            };
+            i = i + 1;
+        };
+    }
+
+    /// Entry function that can be called by anyone
+    public entry fun resolve(proposal_id: u64, nft_dao: address) acquires Proposals, DAO, ProposalVotingStatistics, RegisteredDAOs {
+        // validate if proposal is ready to resolve
+        let dao = borrow_global_mut<DAO>(nft_dao);
+
+        // assert the proposal voting ended
+        let proposals = borrow_global_mut<Proposals>(nft_dao);
+        let proposal = table::borrow_mut(&mut proposals.proposals, proposal_id);
+        let now = timestamp::now_seconds();
+        assert!(now >= proposal.start_time_sec + dao.voting_duration, error::invalid_argument(EPROPOSAL_ENDED));
+
+        // assert the proposal is unresolved yet
+        assert!(proposal.resolution == PROPOSAL_UNRESOLVED, error::invalid_argument(EPROPOSAL_RESOLVED));
+
+        let proposal_stat = &mut borrow_global_mut<ProposalVotingStatistics>(nft_dao).proposals;
+        let voting_stat = table::borrow_mut(proposal_stat, proposal_id);
+        // validate resolve threshold and result
+        let voted = voting_stat.total_no + voting_stat.total_yes;
+        if (voted < dao.resolve_threshold) {
+            // not sufficient token voted
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+            return
+        };
+        let passed = if (voting_stat.total_yes > voting_stat.total_no) {true} else {false};
+        if (passed) {
+            let accts = &borrow_global<RegisteredDAOs>(@dao_platform).res_acct_to_src_acct;
+            assert!(table::contains(accts, nft_dao), error::not_found(ERESOURCE_ACCT_NOT_EXIST));
+            let src_acct = *table::borrow(accts, nft_dao);
+
+            let function_name = proposal.function_name;
+            if (function_name == string::utf8(b"transfer_fund")) {
+                // This is very dangerous.
+                // We should exploring have server side dynamic compiling and deploying DAO contracts in each DAO's own accounts.
```

Comment:
I don't understand this comment. Can you elaborate? 🙏

Reply:
DAO platform can upgrade contract to use the signer capability to steal money of all DAO resource account without any approval needed.